tools/testrunner/log.go

```diff
@@ -65,3 +66,285 @@ func testOnlyStacktrace(stacktrace string) string {
 	}
 	return res
 }
+
+// alertKind represents a category of high-priority alert detected in test output.
+type alertKind string
+
+const (
+	alertKindDataRace alertKind = "DATA RACE"
+	alertKindPanic    alertKind = "PANIC"
+	alertKindFatal    alertKind = "FATAL"
+)
+
+// alert captures a prominent issue detected from stdout/stderr of test runs.
+type alert struct {
+	Kind    alertKind
+	Summary string
+	Details string
```

Comment:
At first I thought we should keep things simple and not include the detailed logs in the report, but it doesn't seem that complicated and very helpful; so seems good to me now.